service/controller/v5/blobclient/blob_client.go

```diff
@@ -0,0 +1,181 @@
+package blobclient
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-07-01/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+const (
+	resourceNotFoundError = "NotFound"
+	maxRetriesRequests    = 3
+)
+
+type Config struct {
+	ContainerName         string
+	GroupName             string
+	StorageAccountName    string
+	StorageAccountsClient *storage.AccountsClient
+}
+
+type BlobClient struct {
+	containerName         string
+	groupName             string
+	storageAccountName    string
+	storageAccountsClient *storage.AccountsClient
+
+	// containerURL is configured separately from the default
+	// parameters.
+	containerURL azblob.ContainerURL
+}
+
+func (c *BlobClient) Boot(ctx context.Context) error {
```

Comment:
`Boot` is now called in `GetCurrentState`, `ApplyCreateChange` and `ApplyUpdateChange`. Basically twice in each loop, causing the resource to block longer than necessary and creating more pressure on the system. Let's make it to be called once. I would also question why we have to do this over and over again at runtime.